locked-asset/factory/src/energy.rs

```diff
@@ -0,0 +1,205 @@
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+use common_structs::Epoch;
+
+#[derive(TypeAbi, TopEncode, TopDecode, NestedEncode, NestedDecode, Clone, PartialEq, Debug)]
+pub struct Energy<M: ManagedTypeApi> {
+    amount: BigInt<M>,
+    last_update_epoch: Epoch,
+    total_locked_tokens: BigUint<M>,
+}
+
+impl<M: ManagedTypeApi> Default for Energy<M> {
+    fn default() -> Self {
+        Self {
+            amount: BigInt::zero(),
+            last_update_epoch: 0,
+            total_locked_tokens: BigUint::zero(),
+        }
+    }
+}
+
+impl<M: ManagedTypeApi> Energy<M> {
+    #[inline]
+    pub fn new(
```

Comment:
energy is used ONLY when unlocking LKMEX v1 - I suppose ?

Reply:
Yes, it is only used to update energy after unlock. At lock, energy is not updated.